HW_5/HW05/HW05.Task4/Program.cs

```diff
@@ -0,0 +1,148 @@
+using System.Collections.Generic;
+using System.Text;
+
+namespace HW05.Task4
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            // input string
+            string listOfAllClients = "123 Main Street St. Louisville OH 43071,432 Main Long Road St. Louisville OH 43071,786 High Street Pollocksville NY 56432,"
++ "54 Holy Grail Street Niagara Town ZP 32908,3200 Main Rd. Bern AE 56210,1 Gordon St. Atlanta RE 13000,"
++ "10 Pussy Cat Rd. Chicago EX 34342,10 Gordon St. Atlanta RE 13000,58 Gordon Road Atlanta RE 13000,"
++ "22 Tokyo Av. Tedmondville SW 43098,674 Paris bd. Abbeville AA 45521,10 Surta Alley Goodtown GG 30654,"
++ "45 Holy Grail Al. Niagara Town ZP 32908,320 Main Al. Bern AE 56210,14 Gordon Park Atlanta RE 13000,"
++ "100 Pussy Cat Rd. Chicago EX 34342,2 Gordon St. Atlanta RE 13000,5 Gordon Road Atlanta RE 13000,"
++ "2200 Tokyo Av. Tedmondville SW 43098,67 Paris St. Abbeville AA 45521,11 Surta Avenue Goodtown GG 30654,"
++ "45 Holy Grail Al. Niagara Town ZP 32918,320 Main Al. Bern AE 56215,14 Gordon Park Atlanta RE 13200,"
++ "100 Pussy Cat Rd. Chicago EX 34345,2 Gordon St. Atlanta RE 13222,5 Gordon Road Atlanta RE 13001,"
++ "2200 Tokyo Av. Tedmondville SW 43198,67 Paris St. Abbeville AA 45522,11 Surta Avenue Goodville GG 30655,"
++ "2222 Tokyo Av. Tedmondville SW 43198,670 Paris St. Abbeville AA 45522,114 Surta Avenue Goodville GG 30655,"
++ "2 Holy Grail Street Niagara Town ZP 32908,3 Main Rd. Bern AE 56210,77 Gordon St. Atlanta RE 13000";
+
+            string zipCode = Console.ReadLine();
+
+            string answer = Travel(listOfAllClients, zipCode);
+
+            Console.WriteLine(answer);
+
+            Console.ReadLine();
+        }
+
+        /// <summary>
+        /// displays all addresses according to the code
+        /// </summary>
+        /// <param name="listOfAllClients">list of addresses of all clients in a line separated by commas</param>
+        /// <param name="zipCode">zip code, according to which the addresses will be displayed</param>
+        /// <returns>string in the specified format</returns>
+        private static string Travel(string listOfAllClients, string zipCode)
+        {
+            // create an array from a string of addresses, where each element is a separate address
+            string[] addressesArray = listOfAllClients.Split(',');
+
+            //create an empty list for entering addresses containing the desired zip code
+            List<string> necessaryAdresses = new List<string>();
+
+            // iteration over each element of the address array
+            for (int i = 0; i < addressesArray.Length; i++)
+            {
+                // check "if the address contains the entered zip code, then"
+                if (addressesArray[i].EndsWith(zipCode))
+                {
+                    // remove the zip code from the address
+                    addressesArray[i] = addressesArray[i].Remove(addressesArray[i].Length - 9, 9);
```

Comment:
I would replace 9 to zipCode.Length
9 - magic number there. and zip could be changed in future - 10 symbols, for instance